File: 2022/Python/01.py
def generator(file) :
    return [[int(x) for x in item.split("\n")] for item in file.split("\n\n")]

def part_1(input) : 
    return max([sum(x) for x in input])

def part_2(input) : 
    return sum(sorted([sum(item) for item in input])[-3:])

File: 2022/Rust/src/bin/01.rs
aoc2022::main!();

fn generator(input : &str) -> Vec<Vec<u32>>{
    input
        .split("\n\n")
        .map(|r| r.lines().map(|cal| cal.parse().unwrap()).collect())
        .collect()
}

fn part_1(input : Vec<Vec<u32>>) -> u32 {
    input
        .iter()
        .map(|r| r.iter().sum::<u32>())
        .max()
        .unwrap()
}

fn part_2(input : Vec<Vec<u32>>) -> u32 {
    input
        .iter()
        .map(|r| r.iter().sum::<u32>())
        .sorted_by(|a,b| b.cmp(a))
        .take(3)
        .sum()    
}